### PR TITLE
Add hack script to update existing core meta table sample example genes

### PR DESCRIPTION
**Pull request by ens-LCampbell**
Add new quick patch script to update meta table sample gene meta info.

The script attempts to quickly but considerately update meta example info on a input set of cores or input of cores + hand picked example gene (stable_id).

The script first selects a sub sample of stable ids per core (n=25). Then checks if these genes have associated homologies in the appropraite compara db on the same host.

As soon as a gene with homology is found, it then moves to create a patch on that basis. 

Users can provide a list of cores and run in auto mode, or a list of cores+stable_ids they hand picked and run in manual mode. 

Script creates some log into, including TSV of core+gene+description of new eg gene within output patch SQL file.

### CODE REVIEW

**Comment by ens-LCampbell** [comment]
This is not a final solution to this problem, but it provides a quick way to create a patch for cores with bad or inappropriate example genes. Python would be better IMO but until we have a discussion this could wait. Since I made the script to make a patch for staging Im sharing the code here. Thanks

File: scripts/misc_scripts/GenerateExampleGenes.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+# limitations under the License.
+
+#!/usr/bin/env bash
+
+INPUT_CORES=$1
+MODE=${2^^}
+PATCH_OUTFILE="${PWD}/$3"
+HOST=$4
+OUTPUT_TSV_SUMMARY="${PWD}/New_EG_Genes_Selected.tsv"
+COMPARA_DB=""
```
Comment (vsitnik) [inline scripts/misc_scripts/GenerateExampleGenes.sh:22]: should it be passed as a parameter? Or only via env?

Comment (vsitnik) [reply scripts/misc_scripts/GenerateExampleGenes.sh:22]: HOST for compara can be different from the HOST  of the cores?
Or are we dealing only with staging?

File: scripts/misc_scripts/GenerateExampleGenes.sh [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+# STABLE_ID_SQL_FILTER="and description NOT LIKE '%uncharacterized%' "
+# STABLE_ID_SQL_FILTER="and description IS NOT NULL "
+
+# Get meta data on core + selected stable_id
+gather_gene_meta () {
+    local PROD=$1
+    local INPUT_STABLE_ID=$2
+
+    echo "Gathering meta data on stable_id: $INPUT_STABLE_ID"
+    $HOST -D $CORE -Ne "select stable_id, seq_region_id, seq_region_start, seq_region_end, description from gene where stable_id=\"$INPUT_STABLE_ID\";" \
```
Comment (vsitnik) [inline scripts/misc_scripts/GenerateExampleGenes.sh:47]: can we split SQL requests into several lines? especially for the long ones?

File: scripts/misc_scripts/GenerateExampleGenes.sh [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        echo -e -n "\nAuto patching : $CORE\n"
+        $HOST -D $CORE -Ne  "select stable_id from gene where biotype = \"$BIOTYPE\" ${STABLE_ID_SQL_FILTER} order by RAND() limit 25;" \
+            | tr "\n" " " >> $META_OUTDIR/rand_genes_${CORE}.txt
+        
+        NEW_GENE=""
+        
+        for NEW_GENE_CHECK in $(tail -n 1 $META_OUTDIR/rand_genes_${CORE}.txt);
+            do
+            # Now check if compara DB contains homologies on randomly select gene stable_id
+            CHECK_COMPARA_HOMOLOGY=`$HOST -D $COMPARA_DB -Ne "SELECT count(*) FROM gene_member JOIN genome_db USING(genome_db_id) JOIN gene_member_hom_stats USING(gene_member_id) WHERE genome_db.name = \"${PROD_NAME}\" AND gene_trees > 0 and stable_id = \"${NEW_GENE_CHECK}\";"`
```
Comment (vsitnik) [inline scripts/misc_scripts/GenerateExampleGenes.sh:79]: Imho `$()` is more visible then `\`\``, but that's rather a personal stuff

**Review by vsitnik — APPROVED** [review]
Just a few minor comments. Feel free to merge though. Thanks

File: scripts/misc_scripts/GenerateExampleGenes.sh (hunk 1 above)
Comment (ens-LCampbell) [reply scripts/misc_scripts/GenerateExampleGenes.sh:22]: I wrote it for staging so I think its fine as is. But let me address

File: scripts/misc_scripts/GenerateExampleGenes.sh (hunk 2 above)
Comment (ens-LCampbell) [reply scripts/misc_scripts/GenerateExampleGenes.sh:47]: In my experience it always lends to issues. Feel free to test it but I don't want to spend more time on this small change

File: scripts/misc_scripts/GenerateExampleGenes.sh (hunk 3 above)
Comment (ens-LCampbell) [reply scripts/misc_scripts/GenerateExampleGenes.sh:79]: Agreed, I shall leave as is if thats ok

File: scripts/misc_scripts/GenerateExampleGenes.sh (hunk 2 above)
Comment (vsitnik) [reply scripts/misc_scripts/GenerateExampleGenes.sh:52]: leave as is then. Thanks!

**Review by vsitnik — APPROVED** [review]
Thanks!

**Comment by ens-LCampbell** [comment]
> Thanks!

Thanks Vasily

**Pull request merged**